docs/dunst.pod

```diff
@@ -444,6 +444,12 @@ WM_CLASS). There should be no need to modify this setting for regular use.
 Display a notification on startup. This is usually used for debugging and there
 shouldn't be any need to use this option.
 
+=item B<verbosity> (values: 'info', 'mesg', 'warn', 'crit', default 'mesg')
```

Comment:
Forgot to add `debug` here.

Reply:
No, this was intentional. IMO debug shouldn't contain any information, which are usable in user contexts, so I guess this is better to hide this from the manpage.

Reply:
I disagree here, even if it's not usable in a user context doesn't mean that someone can't get some insight on what's wrong by setting this value to debug as an experiment. Additionally, the manpage isn't just for users, potential contributors might be trying to find out how to enable debug logging to figure out an issue they've been tracing I don't think it's reasonable or expected to have to look to some other piece of documentation or dig the code to find out there's an even higher level of verbosity.

Reply:
I still think we've got enough log levels for users. I think the debug level should only contain information, which is useful in combination with the source code. And, if you're going to develop on dunst's code, I hope you've read more than just the manpage. Also, if your only piece of information is the manpage, you won't have the source of dunst and therefore won't understand the log.

I'll quote `HACKING.md`:

>  - `LOG_I` (INFO):
>     - Mostly unneccessary info, but important to debug (as the user) some use cases.
>     - e.g.: print the notification contents after arriving
> - `LOG_D` (DEBUG):
>     - Only important during development or tracing some bugs (as the developer).

I have to decline your argument about potential contributors reading the manpage. I think it's more powerful, to assemble a short document, which describes the project's internals, which **is bundled with the code**. And I think polishing `HACKING.md` fits perfect for this.

Reply:
> I still think we've got enough log levels for users.

This is where I lose your line of reasoning. I don't think it's sensible to divide values of options into "for users" and "for developers" especially if we're already including the `verbosity` setting in the docs. I believe it's even worse to purposefully hide a possible value of it because we assume it'll be useless to users.

While only someone who knows the codebase of dunst well will use it routinely, if someone is experiencing an issue and wants to do a quick trace to look what's up the most detailed debugging option is usually the first thing to look for in such cases. And even better it can prompt users to actually post a debug log on relevant issues if they know it shows more details into an issue they've been hunting. And even if not, it's a debug mode, the worse that can come from misusing it is slightly increased disk space usage!

Additionally as I've said above it might also confuse potential contributors since the man-page is the usual quick-reference to settings to look at what values it can take and even with the current `HACKING.md` file there is no specific mention that verbosity can be set to `debug`, only that the debug log level exists and is enabled on debug builds.

Reply:
My intention was (and still is) to prevent, that the debug log becomes a daily driver for users.

To not list it in the manpage/help/etc was my solution. But you convinced me. It's actually an antipattern. And also an analogous form of "security by obscurity".

I think we still should look forward to prevent this, but there's a much cleaner way: Simply putting the right log messages into the right log level.

---

I've pushed a fixup commit for these docs now.